File: XTrailsJPA/src/main/java/com/mattaldrete/enitities/Playlist.java
package com.mattaldrete.enitities;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Entity
@Table(name = "playlist")
public class Playlist {

	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private Integer id;

	private String name;
	private String description;
	private String hashtags;

	@ManyToOne
	@JoinColumn(name = "User_id")
	private User user;

	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public String getHashtags() {
		return hashtags;
	}

	public void setHashtags(String hashtags) {
		this.hashtags = hashtags;
	}

	public User getUser() {
		return user;
	}

	public void setUser(User user) {
		this.user = user;
	}

	@Override
	public String toString() {
		return "Playlist [id=" + id + ", name=" + name + ", description=" + description + ", hashtags=" + hashtags
				+ ", user=" + user + "]";
	}

	public Playlist(Integer id, String name, String description, String hashtags, User user) {
		super();
		this.id = id;
		this.name = name;
		this.description = description;
		this.hashtags = hashtags;
		this.user = user;
	}

	public Playlist() {

	}

}

File: ngXTrails/src/app/entities/playlist/playlist.ts
import { User } from '../user/user';
import { Trail } from '../trail/trail';

export class Playlist {
  // F I E L D S
  id: number;
  name: string;
  description: string;
  hashtags: string;
  trails: Trail[];
  user: User;

  constructor(
    id?: number,
    name?: string,
    description?: string,
    hashtags?: string,
    trails?: Trail[],
    user?: User
  ) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.hashtags = hashtags;
    this.trails = trails;
    this.user = user;
  }

}
